compiler/include/expr.h

```diff
@@ -222,6 +227,41 @@ class CallExpr : public Expr {
   bool            isPrimitive(const char*  primitiveName)                const;
 };
 
+// For storing several call expressions, where
+// choosing between them depends on context
+// (and that choice might need to be done later in resolution).
+// These should only exist during resolution.
```

Comment:
I take the same point of view about cullOverReferences and am even inclined to lump in
callDestructors (well portions of it at least).




On Feb 3, 2016, at 5:42 AM, Michael Ferguson <notifications@github.com<mailto:notifications@github.com>> wrote:


In compiler/include/expr.h<https://secure-web.cisco.com/11gczCWvPgdBVFlY49KkhEO-RAxI0V5gI1I1zA4Pp5jRpjDvqy_5kBiT39lOBQ9fLc8PUFN2EcpELgKhR5Ekufrmp2lUOYS75NT_aEd0s2_BZDsRviq41eq9gjj2nJR-Ar-rZl3PwvRJ6FJ3Qor_qgmHmjkjV1HQ_L08A6pJNQgS1u0Xwxq8k459guahrnnFpje1NNj5IrUqw_NQRrPPRDwIDZnJUd_ZpBjhmC0TTdnB_oDeBFrWtgoEfyp9J56nLjM1_qHYm8KcnxkBKbZ3lpscpSg4HPQgL227NO1Cn93A/https%3A%2F%2Fgithub.com%2Fchapel-lang%2Fchapel%2Fpull%2F3232#discussion_r51721333>:

> @@ -222,6 +227,41 @@ class CallExpr : public Expr {
>    bool            isPrimitive(const char*  primitiveName)                const;
>  };
>
> +// For storing several call expressions, where
> +// choosing between them depends on context
> +// (and that choice might need to be done later in resolution).
> +// These should only exist during resolution.


I was thinking of resolution as including cullOverReferences. I'll update the comment to be more exact.

—
Reply to this email directly or view it on GitHub<https://secure-web.cisco.com/1LzezfAxDNr-Qb4IlKogC108sVZ8dgpjuP7f5GHVoX4lruFvUjhcBE_WH8Lc_1kjZyCwlLlr_npH-xgNv_f5i_ZQIRFBltsN2Fw2qJyYCRA5TZgLb9Ix-O01vvMwhFl3t8ZwVpJwUmd06y7bfNd_U45CpKmYUVljQLSLblDyz7jiFVdYjeOEGvBOWzINPuJUG_KwoFdJflWBVCSmhiavJJR3F1yuOSckrOX2IP_g9kj9lij6kHP1VK5DqeLqXIeQohdnd_7liHqNR6Sjj6j7sYuW3U2YqjdX5prHM6KhWZAE/https%3A%2F%2Fgithub.com%2Fchapel-lang%2Fchapel%2Fpull%2F3232%2Ffiles#r51721333>.